component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateAuthenticator.java

```diff
@@ -209,6 +215,43 @@ protected void processAuthenticationResponse(HttpServletRequest httpServletReque
         }
     }
 
+    /**
+     * Resolves username of the subject attribute.
+     *
+     * @param identifier            username
+     * @param authenticationContext authentication context
+     *
+     * @return resolved username
+     */
+    private String resolveIdentifier(String identifier, AuthenticationContext authenticationContext)
+            throws AuthenticationFailedException, UserStoreException {
+
+        if (getAuthenticatorConfig().getParameterMap().containsKey(X509CertificateConstants.LOGIN_CLAIM_URIS)) {
+            String[] attributeClaimUris = getAuthenticatorConfig().getParameterMap()
+                    .get(X509CertificateConstants.LOGIN_CLAIM_URIS).split(",");
+
+            String tenantDomain = authenticationContext.getTenantDomain();
+            if (tenantDomain == null || tenantDomain.isEmpty()) {
```

Comment:
shall we use StringUtils isBlank here?

Reply:
Resolved with 7c8906ec6627cace5c57f936fd5e011882862dce